### PR TITLE
Adds Shadow demons

### PR DESCRIPTION
**Pull request by hal9000PR**
## What Does This PR Do
Adds Shadow Demons. A variant of slaughter demons. they die in light, heal in the dark.

They have a grapple spell, which fires out a hand that grabs people and throws them to the demon.

If the grapple hits a wall it pulls the demon to that wall/structure. hitting the grapple will disable the lights on whatever was hit and this refers to ALL lights, even glowshrooms and such. 

They can "bloodcrawl" but instead of blood, it requires dark tiles. YOU WILL STILL TAKE DAMAGE WHILE THE LIGHT WHILEST SHADOW CRAWLING however the damage will be halved.

they heal in the dark. 

## Why It's Good For The Game
more variety is fun. Shadowlings were a cool niche of horror monster in the darkness, I hope that this fills that niche

## Images of changes
see updated sprites in post bellow 

## Testing
long TM, + my own personal testing

## Changelog
:cl:
add: shadow demons, a new midround major event.
/:cl:

### CODE REVIEW

**Comment by Contrabang** [comment]
clickbait title smh, very exicted

**Comment by OctusGit** [comment]
So what is the end goal of this antag? I'm guessing similar to a slaughter demon and killing crew indiscriminately? Just in the dark version.

**Comment by Bm0n** [comment]
This sounds like a simple mob version of a nightmare from TG. Cool!

**Comment by hal9000PR** [comment]
> if this isn't a thing already) will we potentially get a demon heart drop upon killing the shadow demon, much like how we do when we kill a slaughter demon? inheriting the shadow crawl instead of the blood crawl like when you consume a slaughter demons heart.

Yeah sure why not

File: code/game/gamemodes/miniantags/demons/shadow demon/shadow_demon.dm [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	L.apply_damage(40, BRUTE, BODY_ZONE_CHEST)
+
+
+/mob/living/simple_animal/demon/shadow/Initialize(mapload)
+	. = ..()
+	AddSpell(new /obj/effect/proc_holder/spell/fireball/shadow_grappel)
+	AddSpell(new /obj/effect/proc_holder/spell/bloodcrawl/shadow_crawl)
+
+/obj/effect/proc_holder/spell/fireball/shadow_grappel
+	name = "Shadow Grappel"
```
Comment (lewcc) [inline code/game/gamemodes/miniantags/demons/shadow demon/shadow_demon.dm:36]:
```suggestion
	name = "Shadow Grapple"
```

Unless it was intentional, this might be a typo

File: code/game/gamemodes/miniantags/demons/shadow demon/shadow_demon.dm [hunk 2]
```diff
@@ -0,0 +1,73 @@
+/mob/living/simple_animal/demon/shadow
+	name = "shadow demon"
+	desc = "A creature thats barely tangable, you can feel its gaze pierce you."
```
Comment (Miraviel) [inline code/game/gamemodes/miniantags/demons/shadow demon/shadow_demon.dm:3]:
```suggestion
	desc = "A creature that's barely tangible, you can feel its gaze pierce you."
```

Comment (Miraviel) [reply code/game/gamemodes/miniantags/demons/shadow demon/shadow_demon.dm:3]: (Not sure about the pierce/piercing, though, piercing sounds more natural to me)

Comment (GDNgit) [reply code/game/gamemodes/miniantags/demons/shadow demon/shadow_demon.dm:3]: If you're using piercing, it should be something like "piercing through you".

**Comment by hal9000PR** [comment]
new sprites by @McRamon 
image
[dreamseeker_nAvo11Errs.webm](https://user-images.githubusercontent.com/69320440/203878807-48638eee-8790-48b7-96c0-641022fb76e6.webm)
[dreamseeker_UGEC9IUk3F.webm](https://user-images.githubusercontent.com/69320440/203878812-39b2a554-148a-44bc-98c7-9420c693c2fd.webm)

**Comment by hal9000PR** [comment]
UPDATE:
double damage on grab bug fixed, takes half damage from light in shadow crawl, and it properly extinguishes glowshrooms, items on the floor are properly extinguished by grapple hits.

**Comment by hal9000PR** [comment]
ok, updated the clarity of when you are in an are that is too bright. it now updates when you walk rather than every 2 seconds. 

also shadow demons will no longer spawn in the light anymore.

**Comment by hal9000PR** [comment]
DNM so I remember to remove the temporary runtime

**Comment by Daylight2** [comment]
After getting to play this once, here's my thoughts on it:

- The core concept is cool, and I had lots of fun hooking people. Killed around 4 people in total before getting killed.
- Shadow walk is quite strong, but relies very heavily on map knowledge. If you know the area you're in, you can evade larger groups with ease and always get the drop on targets. If you don't, you might find yourself feeling like you're stuck in a small area of maint surrounded by all sides with light, and making the most of your mobility becomes quite difficult. Not necessarily a complaint, just something to be aware of.
- Maintenance doors were very annoying. You can bypass them with shadow walk, but if you want to make yourself escape paths, you have to spend around 10 seconds loudly punching a door, which didn't feel very fun and didn't seem to match the theme of a stealthy assassin antag, at least for me. Maybe give it maints access or a fast door prying ability?
- Lights deal a *lot* of damage, and your only way to disable lights on crew has a long cooldown. This makes engaging even lone targets in maints hazardous, any you can forget about taking on any groups bigger than 1. I ended up dying because the lone clown I hooked in maints turned on his flashlight after getting hooked, and I didn't manage to run away fast enough to avoid it. <br>With the shadow demon only really able to do anything against lone people who willingly dive into maints, I think it should be a bit more capable in dealing with them. Making lights unable to be reactivated for around 10 seconds after being grappled would be ideal, but a longer stun time or more health would work as well. 
- I also noticed that certain lights weren't being turned off by the grapple. I did some testing on a private server afterwards, and among the items I tested, various kinds of zipplo lighters and welding tools would fail to be disabled by the grapple, making people with even one of them in their hands or pockets deal massive damage to you even after being grappled.

In conclusion, I think this is a really fun antag, but it feels a bit too fragile and can be easily countered by having a welding tool and flashlight in your pocket.

File: code/modules/lighting/lighting_atom.dm [hunk 3]
```diff
@@ -54,8 +54,12 @@
 		else
 			light = new/datum/light_source(src, .)
 
-/atom/proc/extinguish_light()
-	return
+/atom/proc/extinguish_light(force = FALSE)
+	if(!light)
+		return // TEMPORARY RUNTIME REMOVE LATER
```
Comment (SteelSlayer) [inline code/modules/lighting/lighting_atom.dm:59]: Still planning on removing this?

Comment (hal9000PR) [reply code/modules/lighting/lighting_atom.dm:59]: yes, when its ready for merge.

File: code/datums/spells/bloodcrawl.dm [hunk 4]
```diff
@@ -31,11 +32,245 @@
...
+	L.notransform = FALSE
+
+/obj/effect/proc_holder/spell/bloodcrawl/proc/phaseout(obj/effect/decal/cleanable/B, mob/living/L)
+
+	if(iscarbon(L) && !block_hands(L))
+		return FALSE
+
+	var/turf/mobloc = get_turf(L)
+	L.notransform = TRUE
+	spawn(0)
```
[the remark this answers is not in the document]
Comment (hal9000PR) [reply code/datums/spells/bloodcrawl.dm]: done

File: code/modules/lighting/lighting_atom.dm (hunk 3 above)
Comment (hal9000PR) [reply code/modules/lighting/lighting_atom.dm:59]: removing

**Review by SteelSlayer — APPROVED** [review]

**Comment by DahkGrel** [comment]
Bug: Shadow hand does not put out active floodlights.
